Update all `build.zig`s to the latest Zig nightly

mochalins opened the pull request:
Currently running into a few `XXX.h` missing `cInclude` issues (seems to be okay when running each lib's individual `build.zig` (even `zig build test`), but problematic when building from root `build.zig`). Still WIP, but more eyes would be appreciated.

btipling commented:
@mochalins I also took a stab at this unknowing you had a PR, mine builds but I don't know if it's the best approach, just leaving a link here if you find it helpful: https://github.com/michal-z/zig-gamedev/compare/main...btipling:zig-gamedev:target_changes?expand=1

mochalins commented:
@hazeycode Root `build.zig` and tests passing now, seems all updates to latest Zig master are good. Tested using version `0.12.0-dev.2139+e025ad7b4`. Rebased on latest `unstable` branch for cleaner history. Would appreciate a PR/comments!

hazeycode commented:
Looks good! Thanks for working on this.

Please could you update the minimum zig version in build.zig and the version specified in README.md to the specific version you have tested with?

mochalins commented:
> Looks good! Thanks for working on this.
> 
> Please could you update the minimum zig version in build.zig and the version specified in README.md to the specific version you have tested with?

Rebased on latest `unstable` and updated versions! Please let me know if there's anything else to be done, thanks.

hazeycode commented on libs/zsdl/build.zig:
```diff
@@ -66,25 +70,27 @@ pub const Package = struct {
             },
             .macos => {
-                exe.addRPath(.{ .path = "@executable_path/Frameworks" });
+                exe.root_module.addRPath(.{ .path = "@executable_path/Frameworks" });
```
```suggestion
               exe.root_module.addRPathSpecial("@executable_path/Frameworks");
```
This fixes a compile error on macOS

hazeycode commented:
Thanks!

btipling commented:
@mochalins sorry I missed your comment, fwiw I was building on Windows 11
